### PR TITLE
Fix redis chat message history message hydration

### LINKED ISSUE #1339: Chat History not beeing retrieved / saved
Hi,
I am wondering if anyone else experience the same issue ?
I had some issues on my project and found out examples were returning an anomaly.

image

Buffer Memory is working fine.
image

I have noticed that the error appeared when model used was ChatOpenAI, but not when it is OpenAI. But in both case I can't seems to be able to save history.

### PR DESCRIPTION
**Pull request by nfcampos**
Closes #1339

### CODE REVIEW

File: langchain/src/stores/message/redis.ts [hunk 1]
```diff
@@ -57,19 +56,7 @@ export class RedisChatMessageHistory extends BaseListChatMessageHistory {
     const orderedMessages = rawStoredMessages
       .reverse()
       .map((message) => JSON.parse(message));
-    const previousMessages = orderedMessages
```
Comment (vowelparrot) [inline langchain/src/stores/message/redis.ts:60]: Ah so it was being assigned empty values before going to the `mapStoredMessagesToChatMessages` :( thanks for catching

**Review by vowelparrot — COMMENTED** [review]
I think dynamoDB also needs fixed!

File: langchain/src/stores/tests/redis.int.test.ts [hunk 2]
```diff
@@ -15,7 +15,13 @@ afterAll(async () => {
   await client.disconnect();
 });
 
-test.skip("Test Redis history store", async () => {
+/**
```
Comment (vowelparrot) [inline langchain/src/stores/tests/redis.int.test.ts:18]: Helpful comment. Thanks!

**Comment by nfcampos** [comment]
@vowelparrot no I tested dynamo and it seems fine, because it has its own custom format for saving (not ideal)

**Review by vowelparrot — APPROVED** [review]

**Pull request merged**